inventory-client/src/components/products/wizard-steps/Step1_BasicInfo.tsx

```
@@ -71,19 +71,108 @@ export function Step1_BasicInfo({
+          if (category.children && category.children.length > 0) {
+            addToMap(category.children);
+          }
+        }
+      }
+    };
+    
+    addToMap(allCategories);
 
-    // 將分組的分類資料扁平化為單一陣列
-    const allCategories = Object.values(categoriesGrouped).flat();
+    // 提取父分類（parent_id 為 null 的分類）
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)